Failing test: X-Pack Security API Integration Tests (PKI).x-pack/platform/test/security_api_integration/tests/pki/pki_auth·ts - security APIs - PKI PKI authentication should properly set cookie and authenticate user

A test failed on a tracked branch

```
JestAssertionError: expect(received).toEqual(expected) // deep equality

- Expected  - 0
+ Received  + 3

  Object {
    "authentication_provider": Object {
      "name": "pki",
      "type": "pki",
    },
    "authentication_realm": Object {
      "name": "pki1",
      "type": "pki",
    },
    "authentication_type": "token",
    "elastic_cloud_user": false,
    "email": null,
    "enabled": true,
    "full_name": null,
    "lookup_realm": Object {
      "name": "pki1",
      "type": "pki",
    },
    "metadata": Object {
      "pki_delegated_by_realm": "reserved",
      "pki_delegated_by_user": "kibana_system",
      "pki_dn": "CN=first_client",
    },
    "profile_uid": Any<String>,
    "roles": Array [
      "kibana_admin",
    ],
+   "token": Object {
+     "managed_by": "elasticsearch",
+   },
    "username": "first_client",
  }
    at Context.<anonymous> (pki_auth.ts:155:33)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at Object.apply (wrap_function.js:74:16) {
  matcherResult: {
    actual: {
      username: 'first_client',
      roles: [Array],
      full_name: null,
      email: null,
      token: [Object],
      metadata: [Object],
      enabled: true,
      authentication_realm: [Object],
      lookup_realm: [Object],
      authentication_type: 'token',
      authentication_provider: [Object],
      elastic_cloud_user: false,
      profile_uid: 'u_dGz2ocbp-iLBSRl-53fX_SeDTMxQemXXcZ9shOYlnxY_0'
    },
    expected: {
      username: 'first_client',
      roles: [Array],
      full_name: null,
      email: null,
      enabled: true,
      metadata: [Object],
      authentication_realm: [Object],
      lookup_realm: [Object],
      authentication_provider: [Object],
      authentication_type: 'token',
      elastic_cloud_user: false,
      profile_uid: [Any]
    },
    message: '\x1B[2mexpect(\x1B[22m\x1B[31mreceived\x1B[39m\x1B[2m).\x1B[22mtoEqual\x1B[2m(\x1B[22m\x1B[32mexpected\x1B[39m\x1B[2m) // deep equality\x1B[22m\n' +
      '\n' +
      '\x1B[32m- Expected  - 0\x1B[39m\n' +
      '\x1B[31m+ Received  + 3\x1B[39m\n' +
      '\n' +
      '\x1B[2m  Object {\x1B[22m\n' +
      '\x1B[2m    "authentication_provider": Object {\x1B[22m\n' +
      '\x1B[2m      "name": "pki",\x1B[22m\n' +
      '\x1B[2m      "type": "pki",\x1B[22m\n' +
      '\x1B[2m    },\x1B[22m\n' +
      '\x1B[2m    "authentication_realm": Object {\x1B[22m\n' +
      '\x1B[2m      "name": "pki1",\x1B[22m\n' +
      '\x1B[2m      "type": "pki",\x1B[22m\n' +
      '\x1B[2m    },\x1B[22m\n' +
      '\x1B[2m    "authentication_type": "token",\x1B[22m\n' +
      '\x1B[2m    "elastic_cloud_user": false,\x1B[22m\n' +
      '\x1B[2m    "email": null,\x1B[22m\n' +
      '\x1B[2m    "enabled": true,\x1B[22m\n' +
      '\x1B[2m    "full_name": null,\x1B[22m\n' +
      '\x1B[2m    "lookup_realm": Object {\x1B[22m\n' +
      '\x1B[2m      "name": "pki1",\x1B[22m\n' +
      '\x1B[2m      "type": "pki",\x1B[22m\n' +
      '\x1B[2m    },\x1B[22m\n' +
      '\x1B[2m    "metadata": Object {\x1B[22m\n' +
      '\x1B[2m      "pki_delegated_by_realm": "reserved",\x1B[22m\n' +
      '\x1B[2m      "pki_delegated_by_user": "kibana_system",\x1B[22m\n' +
      '\x1B[2m      "pki_dn": "CN=first_client",\x1B[22m\n' +
      '\x1B[2m    },\x1B[22m\n' +
      '\x1B[2m    "profile_uid": Any<String>,\x1B[22m\n' +
      '\x1B[2m    "roles": Array [\x1B[22m\n' +
      '\x1B[2m      "kibana_admin",\x1B[22m\n' +
      '\x1B[2m    ],\x1B[22m\n' +
      '\x1B[31m+   "token": Object {\x1B[39m\n' +
      '\x1B[31m+     "managed_by": "elasticsearch",\x1B[39m\n' +
      '\x1B[31m+   },\x1B[39m\n' +
      '\x1B[2m    "username": "first_client",\x1B[22m\n' +
      '\x1B[2m  }\x1B[22m',
    name: 'toEqual',
    pass: false
  }
}
```

First failure: [kibana-elasticsearch-snapshot-verify - main](https://buildkite.com/elastic/kibana-elasticsearch-snapshot-verify/builds/7876#019d7541-12d0-49c4-9c00-8f6a2cfd6a44)

<!-- kibanaCiData = {"failed-test":{"test.class":"X-Pack Security API Integration Tests (PKI).x-pack/platform/test/security_api_integration/tests/pki/pki_auth·ts","test.name":"security APIs - PKI PKI authentication should properly set cookie and authenticate user","test.failCount":48}} -->

--- a/x-pack/platform/test/security_api_integration/tests/http_bearer/access_token.ts
+++ b/x-pack/platform/test/security_api_integration/tests/http_bearer/access_token.ts
@@ -43,7 +43,9 @@ export default function ({ getService }: FtrProviderContext) {
         .get('/internal/security/me')
         .set('kbn-xsrf', 'true')
         .set('authorization', `Bearer ${accessToken}`)
-        .expect(200, expectedUser);
+        .expect(200);
+
+      jestExpect(response.body).toMatchObject(expectedUser);
 
       // Make sure we don't automatically create a session
       expect(response.headers['set-cookie']).to.be(undefined);
@@ -53,18 +55,20 @@ export default function ({ getService }: FtrProviderContext) {
       const { accessToken, expectedUser } = await createToken();
 
       // try it once
-      await supertest
+      const responseOne = await supertest
         .get('/internal/security/me')
         .set('kbn-xsrf', 'true')
         .set('authorization', `Bearer ${accessToken}`)
-        .expect(200, expectedUser);
+        .expect(200);
+      jestExpect(responseOne.body).toMatchObject(expectedUser);
 
       // try it again to verity it isn't invalidated after a single request
-      await supertest
+      const responseTwo = await supertest
         .get('/internal/security/me')
         .set('kbn-xsrf', 'true')
         .set('authorization', `Bearer ${accessToken}`)
-        .expect(200, expectedUser);
+        .expect(200);
+      jestExpect(responseTwo.body).toMatchObject(expectedUser);
     });
 
     it('rejects invalid access token via authorization Bearer header', async () => {

--- a/x-pack/platform/test/security_api_integration/tests/kerberos/kerberos_login.ts
+++ b/x-pack/platform/test/security_api_integration/tests/kerberos/kerberos_login.ts
@@ -152,7 +152,7 @@ export default function ({ getService }: FtrProviderContext) {
           .set('Cookie', sessionCookie.cookieString())
           .expect(200);
 
-        jestExpect(spnegoResponse.body).toEqual({
+        jestExpect(spnegoResponse.body).toMatchObject({
           username: 'tester@TEST.ELASTIC.CO',
           roles: expectedUserRoles,
           full_name: null,

--- a/x-pack/platform/test/security_api_integration/tests/pki/pki_auth.ts
+++ b/x-pack/platform/test/security_api_integration/tests/pki/pki_auth.ts
@@ -152,7 +152,7 @@ export default function ({ getService }: FtrProviderContext) {
         .set('Cookie', sessionCookie.cookieString())
         .expect(200);
 
-      jestExpect(response.body).toEqual({
+      jestExpect(response.body).toMatchObject({
         username: 'first_client',
         roles: ['kibana_admin'],
         full_name: null,

